pvlib/snowcoverage.py

```diff
@@ -0,0 +1,170 @@
+              sliding_coefficient=0.197):
+    '''
+    Calculates the fraction of the slant height of a row of modules covered by
+    snow at every time step.
+
+    Implements the model described in [1]_ with minor improvements in [2]_,
+    with the change that the output is in fraction of the row's slant height
+    rather than in tenths of the row slant height. As described in [1]_, model
+    validation focused on fixed tilt systems.
+
+    Parameters
+    ----------
+    snowfall : Series
+        Accumulated snowfall within each time period. [cm]
+    poa_irradiance : Series
+        Total in-plane irradiance [W/m^2]
+    temp_air : Series
+        Ambient air temperature at the surface [C]
+    surface_tilt : numeric
+        Tilt of module's from horizontal, e.g. surface facing up = 0,
+        surface facing horizon = 90. Must be between 0 and 180. [degrees]
+    initial_coverage : float, default None
+        Fraction of row's slant height that is covered with snow at the
+        beginning of the simulation. If None (default) then the initial
+        coverage is set to the snowfall in the first time period. [unitless]
+    threshold_snowfall : float, default 1.0
+        Hourly snowfall above which snow coverage is set to the row's slant
+        height. [cm/hr]
+    m : float, default 80.
+        Coefficient used in [1]_ to determine if snow can slide given
+        irradiance and air temperature. [W/(m^2 C)]
+    sliding coefficient : float, default 0.197
+        Empirical coefficient used in [1]_ to determine how much
+        snow slides off in each time period. [unitless]
+
+    Returns
+    -------
+    snow_coverage : Series
+        The fraction of the slant height of a row of modules that is covered
+        by snow at each time step.
+
+    References
+    ----------
+    .. [1] Marion, B.; Schaefer, R.; Caine, H.; Sanchez, G. (2013).
+       "Measured and modeled photovoltaic system energy losses from snow for
+       Colorado and Wisconsin locations." Solar Energy 97; pp.112-121.
+    .. [2] Ryberg, D; Freeman, J. (2017). "Integration, Validation, and
+       Application of a PV Snow Coverage Model in SAM" NREL Technical Report
+       NREL/TP-6A20-68705
+    '''
+
+    # set up output Series
+    snow_coverage = pd.Series(index=poa_irradiance.index, data=np.nan)
+    if initial_coverage is None:
+        initial_coverage = 0.
+
+    # determine amount that snow can slide in each timestep
+    can_slide = temp_air > poa_irradiance / m
+    slide_amt = sliding_coefficient * sind(surface_tilt) * \
+        _time_delta_in_hours(poa_irradiance.index)
+
+    uncovered = pd.Series(0.0, index=poa_irradiance.index)
+    uncovered[can_slide] = slide_amt[can_slide]
+
+    # build list of intervals between snow fall events
+    snow_events = snowfall[snow_nrel_fully_covered(snowfall,
+                                                   threshold_snowfall)].index
```

Comment:
Hard to follow a function called within an indexer. It will also make the traceback a lot more complicated if there's an exception.